core/import_data/import_records_section_C.py

```diff
@@ -0,0 +1,196 @@
+                )
+
+            # try to complete some missing data from previous year report
+            if previous_year_price:
+                if previous_year_prices_obj:
+                    if previous_year_prices_obj.current_year_price:
+                        if (
+                            previous_year_prices_obj.current_year_price
+                            != previous_year_price
+                        ):
+                            logger.warning(
+                                f"⚠️  [row: {index_row + OFFSET}]"
+                                f"[country: {current_country['obj'].name}][substance: {chemical_name}]"
+                                f"[year: {report_details['year']}] Mismatch in price declaration."
+                            )
+                    else:
+                        previous_year_prices_obj.current_year_price = (
+                            previous_year_price
+                        )
+                        previous_year_prices_obj.save()
+                else:
+                    previous_year_cp_report = get_cp_report(
+                        report_details["year"] - 1,
+                        current_country["obj"].name,
+                        current_country["obj"].id,
+                    )
+                    prev_record_data = {
+                        "country_programme_report_id": previous_year_cp_report.id,
+                        "substance": substance,
+                        "blend": blend,
+                        "previous_year_price": None,
+                        "previous_year_text": "",
+                        "current_year_price": previous_year_price,
+                        "current_year_text": row[report_details["previous"]],
+                        "remarks": "",
+                        "display_name": chemical_name,
+                        "source_file": FILE_NAME,
+                    }
+                    CountryProgrammePrices.objects.create(**prev_record_data)
+
+            # create prices record
+            record_data = {
+                "country_programme_report_id": cp_report.id,
+                "substance": substance,
+                "blend": blend,
+                "previous_year_price": previous_year_price,
+                "previous_year_text": row[report_details["previous"]],
+                "current_year_price": current_year_price,
+                "current_year_text": current_year_text,
+                "remarks": row[report_details["remarks"]],
+                "display_name": chemical_name,
+                "source_file": FILE_NAME,
+            }
+
+            previous_year_prices_obj = CountryProgrammePrices.objects.create(
+                **record_data
+            )
+    logger.info("✔ sheet parsed")
+
+
+def parse_file(file_path):
+    all_sheets = pd.read_excel(file_path, sheet_name=None)
+    for sheet_name, df in all_sheets.items():
+        if sheet_name.strip() == "Section C":
```

Comment:
ar fi bine sa stergem record-urile vechi care provin din acelasi fisier. Din cate stiu fisierele inca nu sunt in varianta finala asa ca o sa tot testam si importam date din fisiere cu acelasi nume. Ideal ar fi sa nu avem duplicate din cauza importurilor multiple.
In utils e metoda delete_old_data care sterge toate datele cu un anumit source_file

Reply:
well, cazul pe care il descrii tu mai sus era tratat 10 linii mai jos unde sterg toate instantele de CountryProgrammePrices existente in baza de date.
daaaaaar, avand in vedere ca din ce am inteles acest model va fi folosit si la alte importuri, atunci da. are sens sa stergem doar datele importante din fisierul parsat aici.